src/app/admin/common/StyledDataGrid.tsx

```diff
@@ -0,0 +1,35 @@
+import { DataGrid, DataGridProps } from "@mui/x-data-grid";
+import { styled } from "styled-components";
+import React from "react";
+
+const DataGridStyling = styled(DataGrid)`
+    & > div {
+        border-radius: 1rem;
+        border: 0px;
+    }
+
+    & .MuiDataGrid-columnHeaders {
+        background-color: ${(props) => props.theme.main.background[2]};
+        border-color: ${(props) => props.theme.main.border};
+        text-align: start;
+        font-size: 1rem;
+    }
+
+    & .datagrid-row-even {
+        text-align: start;
+        font-size: 1rem;
+        background-color: ${(props) => props.theme.main.background[1]};
+        color: ${(props) => props.theme.main.foreground[2]};
+    }
+
+    & .datagrid-row-odd {
+        text-align: start;
+        font-size: 1rem;
+        background-color: ${(props) => props.theme.main.background[0]};
+        color: ${(props) => props.theme.main.foreground[2]};
+    }
+`;
+
+const StyledDataGrid: React.FC<DataGridProps> = (props) => <DataGridStyling {...props} />;
```

Comment:
is this not equivalent?
```suggestion
const StyledDataGrid = styled(DataGrid)`
    & > div {
        border-radius: 1rem;
        border: 0px;
    }

    & .MuiDataGrid-columnHeaders {
        background-color: ${(props) => props.theme.main.background[2]};
        border-color: ${(props) => props.theme.main.border};
        text-align: start;
        font-size: 1rem;
    }

    & .datagrid-row-even {
        text-align: start;
        font-size: 1rem;
        background-color: ${(props) => props.theme.main.background[1]};
        color: ${(props) => props.theme.main.foreground[2]};
    }

    & .datagrid-row-odd {
        text-align: start;
        font-size: 1rem;
        background-color: ${(props) => props.theme.main.background[0]};
        color: ${(props) => props.theme.main.foreground[2]};
    }
`;
```

Reply:
I had hoped so, but styled can't infer generic types, whereas the React FC can. https://stackoverflow.com/questions/69674707/how-to-make-styled-components-generic-to-receive-parameters
(DataGrid has the type parameter GridRowModel or something like that)

Reply:
fair enough!